### PR TITLE
docs update for 1.0.0 and conversion from Sphinx ReST to MyST

### PR DESCRIPTION
**Pull request by grinner**
Docs updated for new changes for primer3 2.6.1 migration

Migrated docs to [MyST](https://myst-parser.readthedocs.io/en/latest/index.html) for sphinx Markdown.  
All `*.rst` files are now `*.md` files 

Update pass for `README`, `CHANGES`, `quickstart`, `miscellany` and `index` documentation for `1.0.0-staging` branch
    
Updates to `howtobuild.md` and `cython_help.md` for better Markdown formatting
      
`CHANGES` and `changes.md` are now pre-prepended.
    
Sphinx Docs support type hinting via [sphinx-autodoc-typehints](https://github.com/tox-dev/sphinx-autodoc-typehints)

`__version__` string in the  `__init__.py` file is now the source of truth for all of the software.  To be possibly updated with SCM migration

Made it consistent that `primer3-py` supports Python 3.8+ and not 3.7 or Python 2

Added a ReST `include` link to top level `CHANGES` file to get rid of duplicate file for maintenance
that is, `changes.rst` is now `changes_link.md`

 added `.github/workflows/update-docs.yml` to build docs when a new version tag is pushed to `master` branch

line length style updates for `thermoanalysis.pyx`

### CODE REVIEW

File: .github/workflows/update-docs.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: "Push New Tagged Version Docs Build" # Build on a new version tag
+on:
+  - push:
+    branches:
+      - "master"
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+"
```
Comment (benpruitt) [inline .github/workflows/update-docs.yml:7]: Were you able to test run this action? Perhaps w/ a dummy `gh-pages-test` branch?

Comment (benpruitt) [reply .github/workflows/update-docs.yml:7]: We could maybe build in consistent "staging" / test support by making the branch to which the documentation is pushed [conditional on the tag format](https://stackoverflow.com/a/58178121) -- i.e., we could just have a persistent `gh-pages-staging` or `gh-pages-test` branch to which pre-release tag doc builds are pushed (i.e., if the tag is an alpha or beta tag)

File: CHANGES [hunk 2]
```diff
@@ -1,130 +1,144 @@
-Changelog
-=========
+# Changelog
 
-Version 0.1:                                                      March 3, 2014
--------------------------------------------------------------------------------
-- Initial release
+## Version 1.0.0                                               February 10, 2023
```
Comment (benpruitt) [inline CHANGES:3]: You already did the hard work but the right-justified date may be more trouble than its worth in the markdown version (i.e., we could just do `Version 1.0.0 (February 10, 2023)` or something)

File: docs/conf.py [hunk 3]
```diff
@@ -273,7 +285,7 @@
 texinfo_documents = [
     (
         'index', 'primer3-py', u'primer3-py Documentation',
-        u'Ben Pruitt, Nick Conway; Wyss Institute', 'primer3-py', 'One line description of project.',
+        'Ben Pruitt, Nick Conway; Wyss Institute', 'primer3-py', 'One line description of project.',
```
Comment (benpruitt) [inline docs/conf.py:288]: Maybe we need to update `One line description of project`? :)

File: docs/quickstart.md [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+    Note that NN thermodynamics will be used to calculate the Tm of sequences
+    up to 60 bp in length, after which point the following formula will be
+    used::
+
+        Tm = 81.5 + 16.6(log10([mv_conc])) + 0.41(%GC) - 600/length
+
+```
+
+```{eval-rst}
+.. py:function:: calc_hairpin(seq, mv_conc=50.0, dv_conc=0.0, dntp_conc=0.8, \
```
Comment (benpruitt) [inline docs/quickstart.md:55]: Might be a follow up effort to take a pass at updating these docs -- e.g., `calc_hairpin` now has an `output_structure` kwarg.

File: docs/quickstart.md [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+To ensure the git hook is excecuted on every commit.
+
+### Pull Requests
+
+Pull Requests should meet the following requirements:
+
+1. Excellent description describing all changes made. Please use markdown syntax highlighting
+2. Attempt to make the PR 1 commit only. Multiple are OK if it helps illustrate the change better.
+3. Commit messages should describe the changes.
+4. Provided you contact the maintainers in advance, code will review your PR, provide feedback and squash merge your code.
```
Comment (benpruitt) [inline docs/quickstart.md:381]: `code will review your PR` -> `they will review your PR`?

Comment (benpruitt) [reply docs/quickstart.md:381]: On a similar note -- thoughts on adding a CODEOWNERS file to simplify some of the GH permissions situations?

File: docs/quickstart.md [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+
+```bash
+$ pre-commit install
+```
+
+To ensure the git hook is excecuted on every commit.
+
+### Pull Requests
+
+Pull Requests should meet the following requirements:
```
Comment (benpruitt) [inline docs/quickstart.md:376]: I would be in favor of adding a line item for updating and/or adding test coverage for any changes / additions

File: .github/workflows/update-docs.yml (hunk 1 above)
Comment (grinner) [reply .github/workflows/update-docs.yml:7]: I have not tested this. I'll try with a staging branch.

File: CHANGES (hunk 2 above)
Comment (grinner) [reply CHANGES:3]: Yeah, I realized after I did it that it's a zero for effort.  I'll update it

File: docs/quickstart.md (hunk 4 above)
Comment (grinner) [reply docs/quickstart.md:55]: Tbh section of the `quickstart.md` I think needs a refactor.  I'll update this with the new args, but lets reconsider the whole of the docs in a follow up.

File: docs/quickstart.md (hunk 5 above)
Comment (grinner) [reply docs/quickstart.md:381]: will update.  +1 on CODEOWNERS

File: docs/quickstart.md (hunk 6 above)
Comment (grinner) [reply docs/quickstart.md:376]: +1 on test coverage line item

File: CHANGES (hunk 2 above)
Comment (grinner) [reply CHANGES:3]: updated

File: docs/conf.py (hunk 3 above)
Comment (grinner) [reply docs/conf.py:288]: Updated

File: docs/quickstart.md (hunk 5 above)
Comment (grinner) [reply docs/quickstart.md:381]: updated

File: docs/quickstart.md (hunk 4 above)
Comment (benpruitt) [reply docs/quickstart.md:55]: Sounds good to me

File: .github/workflows/update-docs.yml (hunk 1 above)
Comment (grinner) [reply .github/workflows/update-docs.yml:10]: Tested on my fork and confirmed to work with comments in the workflow to describe what GH access TOKEN permissions are required

File: .github/workflows/update-docs.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    steps:
+    - uses: actions/checkout@v3.3.0
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4.4.0
+      with:
+        python-version: '3.10'
+    - name: Install and run sphinx to build the docs
+      run: |
+        python -m pip install --upgrade pip
+        pip install cython myst-parser sphinx sphinx_rtd_theme sphinx-autodoc-typehints
```
Comment (benpruitt) [inline .github/workflows/update-docs.yml:24]: If we don't want to introduce `cython` or `venv`, we could minimally drop these requirements into context-specific `*-requirements.txt` files, like `doc-build-requirements.txt` and `pip install -r` them.

Comment (benpruitt) [reply .github/workflows/update-docs.yml:24]: I'm going to resolve this and add a line item to the effort tracker

**Pull request merged**